enhancements/network/baremetal-networking.md

```diff
@@ -0,0 +1,260 @@
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Many customers still have on-premise infrastructure which includes bare metal, VMware vSphere,
+OpenStack and Red Hat RHV.  These customers would like to use the IPI installation approach while utilizing
+their existing environments. This enhancement would provide networking infrastructure including DNS
+and load balancing required for OpenShift to compute environments that do not provide such services.
+
+## Motivation
+
+### Goals
+
+Install an IPI OpenShift cluster on various on-premise non-cloud platforms that
+provides internal DNS and load balancing that is minimally required for OpenShift
+cluster to run properly.
+
+### Non-Goals
+
+## Proposal
+
+The basis and significant portions for this proposal was taken from existing
+[documentation](https://github.com/openshift/installer/blob/master/docs/design/baremetal/networking-infrastructure.md).
+
+In cluster network infrastructure automates a number
+of components that are handled on other platforms by cloud infrastructure services.
+
+### Load-balanced control plane access
+
+Access to the Kubernetes API (port 6443) from clients both external
+and internal to the cluster should be load-balanced across control
+plane machines.
+
+Access to Ignition configs (port 22623) from clients within the
+cluster should also be load-balanced across control plane machines.
+
+In both cases, the installation process expects these ports to be
+reachable on the bootstrap instance at first and then later on the
+newly-deployed control plane machines.
+
+#### API VIP (Virtual IP)
+
+In cluster network infrastructure, a VIP (Virtual IP) is used to provide
+failover of the API server across the control plane machines
+(including the bootstrap instance). This "API VIP" is provided by the user
+as an `install-config.yaml` [parameter](https://github.com/openshift/installer/blob/master/pkg/types/baremetal/platform.go#L57-L63)
+and the installation process configures `keepalived` to manage this VIP.
+
+The API VIP first resides on the bootstrap instance.
+This `keepalived` [instance](https://github.com/openshift/machine-config-operator/blob/master/manifests/baremetal/keepalived.yaml)
+are run as a [static pod](https://kubernetes.io/docs/tasks/administer-cluster/static-pod/) and the
+[relevant assets](https://github.com/openshift/machine-config-operator/blob/master/manifests/baremetal/keepalived.conf.tmpl#L7)
+are rendered by the Machine Config Operator.
+
+The VIP will move to one of the control plane nodes, but only after the
+bootstrap process has completed and the bootstrap instance is stopped. This happens
```

Comment:
> the bootstrap instance is stopped

this is technically incorrect, the users shouldn't have to shutdown the bootstrap-host to move the API VIP to control-plane, we should be able to communicate to control-plane as soon as it's up..?

This seems like a bug...?

Reply:
cc @russellb

Reply:
@abhinavdahiya the bootstrap vrrp_interface priority is set to 50 and control plane 40.  The only way for the VIP to move to the CP is for the boostrap instance of keepalived to be stopped or the entire machine.  The VIP would be under control of a CP node.  haproxy would then load balance between the other CP nodes